aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPluginTest.kt

```diff
@@ -53,16 +61,35 @@ import org.junit.Test
 class RealAWSCognitoAuthPluginTest {
 
     private var logger = mockk<Logger>(relaxed = true)
-    private var authConfiguration = mockk<AuthConfiguration>()
-    private var authService = mockk<AWSCognitoAuthServiceBehavior>()
+    private val appClientId = "topSecretClient"
+    private var authConfiguration = mockk<AuthConfiguration> {
+        every { userPool } returns UserPoolConfiguration.invoke {
+            this.appClientId = this@RealAWSCognitoAuthPluginTest.appClientId
+        }
+    }
+
+    private val mockCognitoIPClient = mockk<CognitoIdentityProviderClient>()
+    private var authService = mockk<AWSCognitoAuthServiceBehavior> {
+        every { cognitoIdentityProviderClient } returns mockCognitoIPClient
+    }
 
     private var authEnvironment = mockk<AuthEnvironment> {
         every { configuration } returns authConfiguration
         every { logger } returns this@RealAWSCognitoAuthPluginTest.logger
         every { cognitoAuthService } returns authService
     }
 
-    private var authStateMachine = mockk<AuthStateMachine>(relaxed = true)
+    private var currentState: AuthenticationState = AuthenticationState.Configured()
```

Comment:
Should this be test specific, since each API depends on different current state?

Reply:
This is default value. Instead of each tests defining mock machine with every.. , they can just set this object to whatever state they want.